code/modules/reagents/chemistry/holder.dm

```diff
@@ -408,9 +410,11 @@
 	if(can_overdose)
 		if(addiction_tick == 6)
 			addiction_tick = 1
-			for(var/addiction in cached_addictions)
-				var/datum/reagent/R = addiction
-				if(C && R)
+			if(cached_addictions)
+				for(var/addiction in cached_addictions)
+					var/datum/reagent/R = addiction
+					if(!C || !R)
```

Comment:
The `!R` check will never pass, `R` will always exist at this point. It can be safely removed.

`C` should be checked for before, not every loop like this.

Reply:
I assume the `C` check is there in the case they somehow gets deleted in the process (eg. mob is afflicted by something that deletes 'em on death and dies from withdrawl).